designer/client/src/components/graph/PanZoomPlugin.ts

```diff
@@ -1,240 +1,161 @@
+        const center = content.center();
+        const translate = this.getTranslatedCenter(center, this.viewport, scale);
+
+        const zoomTransform = zoomIdentity.translate(translate.x, translate.y).scale(scale);
+
+        this.paperSelection
+            .interrupt(transitionName)
+            .transition(transitionName)
+            .duration(isVisualTesting ? 0 : 750)
+            .call(this.zoomBehavior.transform, zoomTransform);
     }
 
-    private cleanup() {
-        this.panStart = null;
-        this.globalCursor.disable();
+    zoomIn(): void {
+        this.zoomTo(this.zoom * 2);
     }
 
-    fitSmallAndLargeGraphs = debounce((): void => {
-        this.setAnimationClass({ enabled: true });
-        this.instance.zoom(1);
-        this.instance.resize();
-        this.instance.updateBBox();
-        this.instance.fit();
-        const { realZoom } = this.instance.getSizes();
-        const toZoomBy = realZoom > 1.2 ? 1 / realZoom : 0.78; //the bigger zoom, the further we get
-        this.instance.zoomBy(toZoomBy);
-        this.instance.center();
-    }, 100);
-
-    zoomIn = throttle((): void => {
-        this.setAnimationClass({ enabled: true });
-        this.instance.zoomIn();
-    }, 500);
-
-    zoomOut = throttle((): void => {
-        this.setAnimationClass({ enabled: true });
-        this.instance.zoomOut();
-    }, 500);
-
-    private pan = throttle((point: { x: number; y: number }) => {
-        requestAnimationFrame(() => {
-            this.instance.panBy(point);
-        });
-    }, 16 /*~60 pleasant fps*/);
-
-    panToCells = (cells: dia.Cell[], viewport: g.Rect = new g.Rect(this.paper.el.getBoundingClientRect())) => {
-        const localViewport = this.paper.clientToLocalRect(viewport);
-        const cellsBBox = this.paper.model.getCellsBBox(cells).inflate(10, 50);
-
-        if (localViewport.containsRect(cellsBBox)) {
-            return;
-        }
+    zoomOut(): void {
+        this.zoomTo(this.zoom * 0.5);
+    }
 
-        const [top, right, bottom, left] = [
-            -localViewport.topLine().pointOffset(cellsBBox.topMiddle()),
-            -localViewport.rightLine().pointOffset(cellsBBox.rightMiddle()),
-            localViewport.bottomLine().pointOffset(cellsBBox.bottomMiddle()),
-            localViewport.leftLine().pointOffset(cellsBBox.leftMiddle()),
-        ].map((offset) => Math.min(20, Math.max(0, offset) / 20));
-
-        const panOffset = {
-            y: Math.round(top - bottom),
-            x: Math.round(left - right),
-        };
-
-        this.pan(panOffset);
-        requestAnimationFrame(() => {
-            this.panToCells(cells, viewport);
-        });
+    private getTranslatedCenter = (center: g.Point, viewport: Viewport, scale: number) => {
+        const viewportRelativeCenter = viewport.translate(0, -viewport.y).center();
+        return viewportRelativeCenter.translate(center.scale(-scale, -scale));
     };
 
-    private initPinchZooming(paper: dia.Paper) {
```

Comment:
The pinch event doesn't work as expected on the IOS simulator, is it working on a real device? do we change the approach to zooming in the case of touch devices? 

https://github.com/TouK/nussknacker/assets/9945753/f1530314-dc3b-4e14-a213-11c3fc396ce0